kategory-docs/docs/docs/integrations/rx2/README.md

```diff
@@ -4,6 +4,146 @@ title: Rx2
+  AC.runAsync { getSongUrl() }
+
+val songObservable: ObservableKW<Url> = getUsersAsync().ev()
+val songFlowable: FlowableKW<Url> = getUsersAsync().ev()
+```
+
+[`MonadError`]({{ '/docs/typeclasses/monaderror' | relative_url }}) can be used to start a [Monad Comprehension]({{ '/docs/patterns/monadcomprehensions' | relative_url }}) using the method `bindingE`, with all its benefits.
+
+Let's take an example and convert it to a comprehension. We'll create an observable that loads a song from a remote location, and then reports the current play % every 100 milliseconds until the percentage reaches 100%:
+
+```kotlin
+getSongUrlAsync()
+  .map { songUrl -> MediaPlayer.load(songUrl) }
+  .flatMap {
+    val totalTime = musicPlayer.getTotaltime()
+    Observable.interval(100, Milliseconds)
+      .flatMap { 
+        Observable.create { musicPlayer.getCurrentTime() }
+          .subscribeOn(AndroidSchedulers.mainThread())
+          .map { tick -> (tick / totalTime * 100).toInt() }
+      }
+      .takeUntil { percent -> percent >= 100 }
+      .observeOn(Schedulers.immediate())
+  }
+```
+
+When rewritten using `bindingE` it becomes. Note that any unexpected exception, like AritmeticException when totalTime is 0, is automatically caught and wrapped inside the observable. 
+
+```kotlin
+ObservableKW.errorMonad().bindingE {
+  val songUrl = getSongUrlAsync().bind()
+  val musicPlayer = MediaPlayer.load(songUrl)
+  val totalTime = musicPlayer.getTotaltime()
+
+  val end = PublishSubject.create<Unit>()
+  Observable.interval(100, Milliseconds).takeUntil(end).bind()
+
+  val tick = bindIn(UI) { Observable.create { musicPlayer.getCurrentTime() } }
+  val percent = (tick / totalTime * 100).toInt()
+  if (percent >= 100) {
+    end.onNext(Unit)
+  }
+
+  yield(percent)
+}.ev()
+```
+
+### Subscription and cancellation
+
+Observables created with `bindingE` behave the same way regular observables do, including cancellation by disposing the subscription.
+
+```kotlin
+val disposable = 
+  songObservable.value()
+    .subscribe({ Log.d("Song $it") } , { prinln("Error $it") })
+
+disposable.dispose()
+```
+
+Note that [`MonadError`]({{ '/docs/typeclasses/monaderror' | relative_url }}) provides an alternative to `bindingE` called `bindingECancellable` returning a `kategory.Disposable`. Invoking this `Disposable` causes an `InterruptedException` in the chain which needs to be handled by the subscriber. This behavior is similar to cancelling an `rx.Single`.
+
+```kotlin
+val (observable, disposable) = 
+  ObservableKW.monadError().bindingECancellable {
+    val userProfile = Observable.create { getUserProfile("123") }
+    val friendProfiles = userProfile.friends().map { friend ->
+        bindAsync(observableAsync) { getProfile(friend.id) }
+    }
+    yields(listOf(userProfile) + friendProfiles)
+  }
+
+observable.value()
+  .subscribe({ Log.d("Song $it") } , { prinln("Boom! caused by $it") })
+
+disposable()
```

Comment:
Shouldn't be `disposable.dispose()`?

Reply:
Ahh nvm I though it was a Rx-Disposable. But wouldn't this cause some confusion among RxJava developers?